Code/Editor/Util/ModalWindowDismisser.cpp

```diff
@@ -29,37 +28,22 @@ ModalWindowDismisser::~ModalWindowDismisser()
     }
 }
 
-void ModalWindowDismisser::DismissWindows()
-{
-    for (QDialog* dialog : m_windows)
-    {
-        dialog->close();
-    }
-    m_windows.clear();
-    m_dissmiss = false;
-}
-
 bool ModalWindowDismisser::eventFilter(QObject* object, QEvent* event)
 {
     if (QDialog* dialog = qobject_cast<QDialog*>(object))
     {
         if (dialog->isModal())
         {
-            QEvent::Type test = event->type();
-            if (test == QEvent::Show || test == QEvent::WindowActivate)
+            // if we wait until a Paint event arrives, we can be sure the dialog is fully open
+            // and that close() will work correctly.
+            if (event->type() == QEvent::Paint)
```

Comment:
You might be able to key off the `QEvent::Polish`, which should come before the paint but should also be a valid state to close the widget. If that doesn't work, then the `QEvent::Paint` is good too. This will be much more reliable than a timer, thank you!

Reply:
Hi Chris, Polish doesn't work, I'm afraid. It doesn't always get sent.